calculate_statistics/info_statistics.py

```diff
@@ -0,0 +1,242 @@
+import io
+import json
+import os
+import requests
+import settings
+import string
+from datetime import datetime, timedelta
+from performanceplatform.client import DataSet
+
+'''
+InfoStatistics class: this generates the aggregated data for
+the PP's info-statistics dataset. This is used to identify
+pages with high numbers of problem reports and searches.
+It does the following:
+- Fetches data from the PP for all pages with problem reports
+  or searches
+- Aggregates problem reports for smart answers to the level of
+  the starting URL, to aid comparison
+- Initialises a neat output dataset
+- For all URLs with problem reports or searches, fetches data
+  on the number of unique page views
+- Normalise problem reports / searches by the number of unique
+  page views
+- Write output to a local JSON file and to the PP
+'''
+
+
+class InfoStatistics():
+
+    date_format = "%Y-%m-%dT00:00:00Z"
+    date_format_longer = "%Y-%m-%dT00:00:00+00:00"
+
+    def __init__(self, pp_token):
+        self.end_date = datetime.now()
+        self.start_date = self.end_date - timedelta(days=settings.DAYS)
+        self.pp_token = pp_token
+
+    def construct_pp_query(self, dataset, value,
```

Comment:
Generally speaking, it'd be good to separate `InfoStatistics` into a few classes (I can imagine one for dealing with PP querying and posting, one with getting info out of the GOV.UK API, one for dealing with the actual data manipulation). This would allow seeing the architecture of the code at a glance a bit easier.